lib/userFactory.js

```diff
@@ -65,6 +65,38 @@ class UserFactory extends BaseFactory {
     }
 
     /**
+     * Get a user model by name, id, or token
+     * @method get
+     * @param   {Mixed}     config
+     * @param   {String}    [config.token]      Token of the user to look up
+     * @return  {Promise}
+     */
+    get(config) {
+        if (!config.token) {
+            return super.get(config);
+        }
+
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const TokenFactory = require('./tokenFactory');
+        /* eslint-enable global-require */
+        const tokenFactory = TokenFactory.getInstance();
+
+        return tokenFactory.get({ value: config.token })
+            .then((token) => {
+                if (!token) {
+                    return token;
+                }
+
+                token.lastUsed = (new Date()).toISOString();
+
+                return token.update()
+                    .then(() => this.get(token.userId));
```

Comment:
If it's out there it'll also run after the case where we return `null` if the token isn't found, and then it will error when it tries to get `null.userId`.

Moved to the same line as `return token.update()` so that it looks slightly prettier though.